Nightly test failures, gcc/8.2,9.2, SKX arch, debug build: error: ‘_mm512_loadu_epi64’ was not declared in this scope

**Describe the bug**
Nightly tests began failing to compile in gcc/8.2,9.2, SKX arch, debug builds following merge of these commits:

Remove extended_namespace template paramter for SYCLMemoryOrder/Scope ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/266/changes#e832781a3f03332b954a6ee908c3ab933b098583) / [githubweb](https://github.com/kokkos/kokkos/commit/e832781a3f03332b954a6ee908c3ab933b098583))
    Cherry-pick v3.7.02 changelog into develop [ci skip] ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/266/changes#1af1379994e0f23f213bd7d9caeae18a2a2272c9) / [githubweb](https://github.com/kokkos/kokkos/commit/1af1379994e0f23f213bd7d9caeae18a2a2272c9))
    Kokkos: Mark HWLOC as a TriBITS TPL as well (trilinos/Trilinos#11938) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/266/changes#9df5a01a80509c63b14c1c68e1b1aec989159797) / [githubweb](https://github.com/kokkos/kokkos/commit/9df5a01a80509c63b14c1c68e1b1aec989159797))
    SIMD: make binary op tests to test against all data types (#5913) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/266/changes#915c1746696180a8d4e569e60eecef9aaceaa5f8) / [githubweb](https://github.com/kokkos/kokkos/commit/915c1746696180a8d4e569e60eecef9aaceaa5f8))
    Also create symlinks for CMake configuration files to cmake_packages/Kokkos for TriBITS (#6163) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/266/changes#65ffe4c5d94d83555c565ffa27a1616a3d71b009) / [githubweb](https://github.com/kokkos/kokkos/commit/65ffe4c5d94d83555c565ffa27a1616a3d71b009))
    Allow passing a temporary std::vector to partition_space (#6167) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/266/changes#0b7bed5817a7c095df9d8d8409e00d3a4f61fb1b) / [githubweb](https://github.com/kokkos/kokkos/commit/0b7bed5817a7c095df9d8d8409e00d3a4f61fb1b))


**Error message**:
```
01:26:06 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/kokkos/simd/src/Kokkos_SIMD_AVX512.hpp:391:15: error: ‘_mm512_loadu_epi64’ was not declared in this scope
01:26:06      m_value = _mm512_loadu_epi64(ptr);
01:26:06                ^~~~~~~~~~~~~~~~~~
01:26:06 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/kokkos/simd/src/Kokkos_SIMD_AVX512.hpp:391:15: note: suggested alternative: ‘_mm512_load_epi64’
01:26:06      m_value = _mm512_loadu_epi64(ptr);
01:26:06                ^~~~~~~~~~~~~~~~~~
01:26:06                _mm512_load_epi64
01:26:06 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/kokkos/simd/src/Kokkos_SIMD_AVX512.hpp: In member function ‘void Kokkos::Experimental::simd<long int, Kokkos::Experimental::simd_abi::avx512_fixed_size<8> >::copy_to(Kokkos::Experimental::simd<long int, Kokkos::Experimental::simd_abi::avx512_fixed_size<8> >::value_type*, Kokkos::Experimental::element_aligned_tag) const’:
01:26:06 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/kokkos/simd/src/Kokkos_SIMD_AVX512.hpp:395:5: error: ‘_mm512_storeu_epi64’ was not declared in this scope
01:26:06      _mm512_storeu_epi64(ptr, m_value);
01:26:06      ^~~~~~~~~~~~~~~~~~~
01:26:06 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/kokkos/simd/src/Kokkos_SIMD_AVX512.hpp:395:5: note: suggested alternative: ‘_mm512_store_epi64’
01:26:06      _mm512_storeu_epi64(ptr, m_value);
01:26:06      ^~~~~~~~~~~~~~~~~~~
01:26:06      _mm512_store_epi64
01:26:06 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/kokkos/simd/src/Kokkos_SIMD_AVX512.hpp: In member function ‘void Kokkos::Experimental::simd<long unsigned int, Kokkos::Experimental::simd_abi::avx512_fixed_size<8> >::copy_from(const value_type*, Kokkos::Experimental::element_aligned_tag)’:
01:26:06 /home/jenkins/blake-new/workspace/Kokkos_develop_blake_gcc_openmp_pthread_serial_820/kokkos/simd/src/Kokkos_SIMD_AVX512.hpp:522:15: error: ‘_mm512_loadu_epi64’ was not declared in this scope
01:26:06      m_value = _mm512_loadu_epi64(ptr);
01:26:06                ^~~~~~~~~~~~~~~~~~
01:26:06 [ 93%] Building CXX object algorithms/unit_tests/CMakeFiles/Kokkos_AlgorithmsUnitTest_StdSet_C.dir/TestStdAlgorithmsFindEnd.cpp.o
```

**Please also include the following items to support reproducing the bug**
**Reproducer** (Blake testbed):
```bash
module load cmake/3.19.3 gcc/8.2.0

$KOKKOS_PATH/generate_makefile.bash --with-devices=Serial --arch=SKX --compiler=g++ --cxxflags="-g -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized " --cxxstandard="17" --kokkos-path=$KOKKOS_PATH --no-examples   --with-options=enable_large_mem_tests --debug
```

--- a/simd/src/Kokkos_SIMD_AVX512.hpp
+++ b/simd/src/Kokkos_SIMD_AVX512.hpp
@@ -388,11 +388,11 @@ class simd<std::int64_t, simd_abi::avx512_fixed_size<8>> {
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION void copy_from(value_type const* ptr,
                                                        element_aligned_tag) {
-    m_value = _mm512_loadu_epi64(ptr);
+    m_value = _mm512_loadu_si512(ptr);
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION void copy_to(
       value_type* ptr, element_aligned_tag) const {
-    _mm512_storeu_epi64(ptr, m_value);
+    _mm512_storeu_si512(ptr, m_value);
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd operator>>(int rhs) const {
     return _mm512_srai_epi64(m_value, rhs);
@@ -519,11 +519,11 @@ class simd<std::uint64_t, simd_abi::avx512_fixed_size<8>> {
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION void copy_from(value_type const* ptr,
                                                        element_aligned_tag) {
-    m_value = _mm512_loadu_epi64(ptr);
+    m_value = _mm512_loadu_si512(ptr);
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION void copy_to(
       value_type* ptr, element_aligned_tag) const {
-    _mm512_storeu_epi64(ptr, m_value);
+    _mm512_storeu_si512(ptr, m_value);
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd
   operator>>(unsigned int rhs) const {

